cognite/client/_api/files.py

```diff
@@ -562,10 +562,17 @@ def upload_bytes(
             source_modified_time=source_modified_time,
             security_categories=security_categories,
         )
+        try:
+            res = self._post(
+                url_path=self._RESOURCE_PATH, json=file_metadata.dump(camel_case=True), params={"overwrite": overwrite}
+            )
+        except CogniteAPIError as e:
+            if e.code == 403 and "insufficient access rights" in e.message:
+                dsid_notice = " Try to provide a data_set_id." if data_set_id is None else ""
+                msg = f"Could not upload a file due to insufficient access rights.{dsid_notice}"
+                raise CogniteFileUploadError(message=msg, code=e.code) from e
```

Comment:
This is a breaking change. Also not convinced that this should be a CogniteFileUploadError, so don't think it merits a major bump. You can raise a subclass of CogniteAPIError instead, so not to break the API.

Reply:
Good point, I didn't realized it's a breaking change! Replaced with a subclass of API error.

Reply:
I guess `CogniteAuthError` was not used as it doesn't inherit from `CogniteAPIError`, but `CogniteException`.